Fixed #1529. historical bbox added to grail pull modal

bobi7 opened the pull request:
ref #1529

Commit:
Fixed #1529. historical bbox added to grail pull modal

bobi7 commented:
Should i limit the amount of bboxes we keep in the history?

brianhatchl commented:
yah, 5?

Commit:
increase limit for history bbox dropdown to 5. order by latest used

maxgrossman commented on modules/Hoot/tools/grailPull.js:
```diff
@@ -114,6 +114,14 @@ export default class GrailPull {
 
+
+        let history = JSON.parse( Hoot.context.storage('history') );
+        if ( history.bboxHistory.length === 5 ) {
```
I do not think anyone would have be clever and mutate the history to have > 5 entries, but they could. 
So should we do a greater than or equal to? 
And then instead of pop do a slice like `history.bboxHistory = history.bboxHistory.slice(1,5)`.

This ensures in the greater than or equal scenario we reset to first 4 bboxes before we add the new bbox on line 122

bobi7 replied:
true. better to be safe

maxgrossman reviewed (changes requested):
Just the one comment. I also noticed a bug that's outside scope of this ticket but bbox related that I'll put in comments

maxgrossman commented:
If I select derive differential and click use visual extent I get this javascript error

```
selectBbox.js:216 Uncaught TypeError: Cannot read property 'node' of undefined
    at SelectBbox.handleBbox (selectBbox.js:216)
    at HTMLButtonElement.<anonymous> (selectBbox.js:152)
    at HTMLButtonElement.<anonymous> (on.js:27)
```

... the same happens if you draw bbox in derive changeset, maybe we just need to put that dropdown there in derive modal too?

maxgrossman commented on modules/Hoot/tools/selectBbox.js:
```diff
@@ -176,6 +212,10 @@ export default class SelectBbox extends EventEmitter {
         this.minLatInput.property( 'value', this.minlat );
 
+        if ( this.bboxSelectType !== 'boundsHistory' ){
+            this.dropdownContainer.node().value = '';
```
Also, to wear my d3 badge on my chest, can we use some d3 here instead? 
```javascript
d3.select(...).attr('value', '')
```

bobi7 replied:
🤓 um actually...you would use `.property` instead but i know what you meant 🤓  
-@melvin

maxgrossman replied:
lol

maxgrossman commented:
Also, I created this error with these steps...

```
Uncaught TypeError: Cannot read property '0' of undefined
    at SelectBbox.updateCoords (selectBbox.js:60)
    at SelectBbox.handleBbox (selectBbox.js:207)
    at finish (clip_bounding_box.js:24)
    at Dispatch.call (dispatch.js:57)
    at ret (clip.js:118)
    at SVGSVGElement.<anonymous> (clip.js:34)
    at SVGSVGElement.<anonymous> (on.js:27)
```

1. select pull remote data
2. draw a bounds
3. click off the job
4. go back to pull remote data
5. select that bounds from my history
6. go to draw bounds
7. click my first corner

Also the mouse icon switches from the cross-hair to the regular mouse icon.

Commit:
Merge branch 'hoot2x' into 1529

Commit:
added >= check for limit for history instead of ===

Commit:
clearing dropdown the d3 way

Commit:
added bbox history dropdown to create Differential Changeset

fixed bug where selecting bbox for clip was trying to clear the bbox history dropdown which it doesnt actually have

maxgrossman commented:
Big :+1: here - I think when you added dropdown to the differential changesets modal it made the undefined errors above no longer applicable
